app/test_goldens/groups/src/pages/course/create/course_template_page_test.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2024 Sharezone UG (haftungsbeschränkt)
+// Licensed under the EUPL-1.2-or-later.
+//
+// You may obtain a copy of the Licence at:
+// https://joinup.ec.europa.eu/software/page/eupl
+//
+// SPDX-License-Identifier: EUPL-1.2
+
+import 'dart:math';
+
+import 'package:bloc_provider/bloc_provider.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:golden_toolkit/golden_toolkit.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:provider/provider.dart';
+import 'package:sharezone/groups/src/pages/course/create/bloc/course_create_bloc.dart';
+import 'package:sharezone/groups/src/pages/course/create/bloc/course_create_bloc_factory.dart';
+import 'package:sharezone/groups/src/pages/course/create/bloc/my_admin_school_classes_provider.dart';
+import 'package:sharezone/groups/src/pages/course/create/pages/course_template_page.dart';
+import 'package:sharezone_widgets/sharezone_widgets.dart';
+
+import 'course_template_page_test.mocks.dart';
+
+@GenerateNiceMocks([
+  MockSpec<CourseCreateBlocFactory>(),
+  MockSpec<MyAdminSchoolClassesProvider>(),
+  MockSpec<CourseCreateBloc>(),
+])
+void main() {
+  group(CourseTemplatePage, () {
+    late MockCourseCreateBlocFactory courseCreateBlocFactory;
+    late MockCourseCreateBloc courseCreateBloc;
+
+    setUp(() {
+      courseCreateBlocFactory = MockCourseCreateBlocFactory();
+      courseCreateBloc = MockCourseCreateBloc();
+
+      final random = Random(42);
+      when(courseCreateBlocFactory.create(schoolClassId: null))
+          .thenReturn(courseCreateBloc);
+      when(courseCreateBloc.isCourseTemplateAlreadyAdded(any))
+          .thenAnswer((_) => random.nextBool());
```

Comment:
I'm not sure, since you hardcoded the seed, but could this make any trouble in a golden test? Shouldn't this be hardcoded value (true/false) in the test itself?

Reply:
Moved it to the `CourseCreateBloc` 👍

Reply:
This shouldn't be a problem because the idea is more have some course that show the check (currently, we don't care which courses these are because the golden test doesn't test the logic of it). Since the seed is hard-coded, it will always produce the same results